lib/global-styles.php

```diff
@@ -571,9 +569,6 @@ function gutenberg_experimental_global_styles_enqueue_assets() {
  * @return array New block editor settings
  */
 function gutenberg_experimental_global_styles_settings( $settings ) {
-	if ( ! gutenberg_experimental_global_styles_has_theme_json_support() ) {
```

Comment:
In this case, instead of removing the check, how about:

- use the check to determine whether `__experimentalGlobalStylesUserEntityId` and `__experimentalGlobalStylesBase` should be added to settings
- always add `styles` to settings